pkgs/development/python-modules/jupyterlab/default.nix

```diff
@@ -0,0 +1,27 @@
+{ lib, buildPythonPackage, isPy3k, fetchPypi, ipython_genutils, jupyterlab_launcher, notebook }:
+buildPythonPackage rec {
+  pname = "jupyterlab";
+  version = "0.31.12";
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1hp6p9bsr863glildgs2iy1a4l99m7rxj2sy9fmkxp5zhyhqvsrz";
+  };
+
+  propagatedBuildInputs = [
+    ipython_genutils
+    jupyterlab_launcher
+    notebook
+  ];
+
+  # No tests in archive
+  doCheck = false;
```

Comment:
Seems like there are tests in `jupyterlab/tests`. However, it seems like node is required:

> Please install nodejs 5+ and npm before continuing.